index.js

```diff
@@ -1,73 +1,214 @@
-    testCases: [{
-        input: "[1,2,3,4,5]",
-        output: "5"
-    }]
-}];
-
-
-const SUBMISSION = [
-
-]
+  questionId: 1,
+  title: "Two states",
+  description: "Given an array , return the maximum of the array?",
+  testCases: [{
+      input: "[1,2,3,4,5]",
+      output: "5"
+    }
+  ]
+}]
+const SUBMISSIONS = []
+const secretKey = crypto.randomBytes(32).toString('hex')
+
+app.use(express.json())
+// app.use(bodyParser.json())
+
+app.use(session({
+  secret: secretKey,
+  resave: false,
+  saveUninitialized: false,
+  cookie: {
+    httpOnly: true
+  }
+}))
+
+const requireAuth = (req, res, next) => {
+  if (req.session && req.session.isAuth) {
+    return next()
+  } else {
+    return res.redirect('/login')
+  }
+}
+
+app.get('/login', function(req, res){
+  res.send("Login to proceed");
+})
 
-app.post('/signup', function(req, res) {
+app.post('/signup', async function(req, res) {
   // Add logic to decode body
   // body should have email and password
-
-
+  const {email, password, isAdmin} = req.body
+  let isAdminValue = false
+  if(isAdmin === "on"){
+    isAdminValue = true
+  }else{
+    isAdminValue = false
+  }
   //Store email and password (as is for now) in the USERS array above (only if the user with the given email doesnt exist)
-
-
+  const hashedPassword = await bcrypt.hash(password, 7)
+  const statusCode = addUser(email, hashedPassword, isAdminValue)
   // return back 200 status code to the client
-  res.send('Hello World!')
+  res.sendStatus(statusCode)
 })
 
 app.post('/login', function(req, res) {
   // Add logic to decode body
   // body should have email and password
-
+  const { email, password } = req.body
   // Check if the user with the given email exists in the USERS array
   // Also ensure that the password is the same
-
-
+  const isCredientialsMatched = USERS.some(function(user) {
+    if(user.email === email){
+      const isMatched = bcrypt.compareSync(password, user.password)
+      return isMatched
+    }
+  })
   // If the password is the same, return back 200 status code to the client
   // Also send back a token (any random string will do for now)
   // If the password is not the same, return back 401 status code to the client
-
-
-  res.send('Hello World from route 2!')
+  if(isCredientialsMatched){
+    req.session.currentUser = email
+    req.session.isAuth = true
+    const token = crypto.randomBytes(32).toString('hex')
+    res.status(200).json({
+      message: "Hello, Welcome",
+      token: token
+    })
+  } else{
+    return res.sendStatus(401)
+  }
 })
 
-app.get('/questions', function(req, res) {
-
+app.get('/questions', requireAuth, function(req, res) {
   //return the user all the questions in the QUESTIONS array
-  res.send("Hello World from route 3!")
+  res.send(QUESTIONS)
 })
 
-app.get("/submissions", function(req, res) {
-   // return the users submissions for this problem
-  res.send("Hello World from route 4!")
+app.get("/submissions", requireAuth, function(req, res) {
```

Comment:
In the HTTP method `GET` you can not add a body, so the line `const { questionId } = req.body` will return `undefined` and this route will always return `status 404: Not submitted before`.